server/src/main/java/org/opensearch/common/geo/GeoUtils.java

```diff
@@ -95,6 +95,12 @@ public class GeoUtils {
     /** rounding error for quantized latitude and longitude values */
     public static final double TOLERANCE = 1E-6;
 
+    public static final PointParser POINT_PARSER;
+
+    static {
+        POINT_PARSER = new PointParser(LONGITUDE, LATITUDE, true);
```

Comment:
Not able to understand what is benefit of creating a PointParser object here when the whole GeoUtils class has static functions.

Reply:
True for this GeoUtils case. If PointParser object is used in other classes, writing unit tests will be easier by injecting mock of PointParser.

Reply:
but given that point parser class it not maintaining the state we should make the function static. In terms of code reading and number of objects getting created its far better to have static functions which has no states.